src/Cli/Utils.cs

```diff
@@ -898,10 +898,10 @@ public static bool TryMergeConfigsIfAvailable(out string mergedConfigFile)
             {
                 string baseConfigFile = RuntimeConfigPath.DefaultName;
                 string environmentBasedConfigFile = RuntimeConfigPath.GetFileName(environmentValue, considerOverrides: false);
-                mergedConfigFile = RuntimeConfigPath.GetMergedFileNameForEnvironment(CONFIGFILE_NAME, environmentValue);
 
                 if (DoesFileExistInCurrentDirectory(baseConfigFile) && !string.IsNullOrEmpty(environmentBasedConfigFile))
                 {
+                    mergedConfigFile = RuntimeConfigPath.GetMergedFileNameForEnvironment(CONFIGFILE_NAME, environmentValue);
```

Comment:
This line could be moved after the line where the merge function is called.
`Merge(baseConfigJson, overrideConfigJson)`

This way, the merged config file name is set only after the json strings are merged successfully.